substrate/client/authority-discovery/src/worker/addr_cache.rs

```diff
@@ -35,11 +45,70 @@ pub(super) struct AddrCache {
 	peer_id_to_authority_ids: HashMap<PeerId, HashSet<AuthorityId>>,
 }
 
+impl std::fmt::Debug for AddrCache {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		f.debug_struct("AddrCache")
+			.field("authority_id_to_addresses", &self.authority_id_to_addresses)
+			.field("peer_id_to_authority_ids", &self.peer_id_to_authority_ids)
+			.finish()
+	}
+}
+
+impl PartialEq for AddrCache {
+	fn eq(&self, other: &Self) -> bool {
+		self.authority_id_to_addresses == other.authority_id_to_addresses &&
+			self.peer_id_to_authority_ids == other.peer_id_to_authority_ids
+	}
+}
+
+fn write_to_file(path: impl AsRef<Path>, contents: &str) -> io::Result<()> {
+	let path = path.as_ref();
+	let mut file = File::create(path)?;
+
+	#[cfg(target_family = "unix")]
+	{
+		use std::os::unix::fs::PermissionsExt;
+		fs::set_permissions(path, std::fs::Permissions::from_mode(0o600))?;
```

Comment:
Seems weird to care about the permissions for this file.

Reply:
I saw we did [it here](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/client/keystore/src/local.rs#L530) - but maybe not at all needed? Should I remove the whole `#[cfg(target_family = "unix")]` block?

Reply:
I would, this is literally publicly available data scrapped from the internet :D.

Reply:
Thx, removed.